lib/backend/k8s/resources/node.go

```diff
@@ -160,11 +160,16 @@ func (c *nodeClient) EnsureInitialized() error {
 }
 
 func (c *nodeClient) Watch(ctx context.Context, list model.ListInterface, revision string) (api.WatchInterface, error) {
-	if len(list.(model.ResourceListOptions).Name) != 0 {
-		return nil, fmt.Errorf("cannot watch specific resource instance: %s", list.(model.ResourceListOptions).Name)
+	// Build watch options to pass to k8s.
+	opts := metav1.ListOptions{ResourceVersion: revision, Watch: true}
+	name := list.(model.ResourceListOptions).Name
```

Comment:
I see that we didn't check before but should we check that this conversion works?

Reply:
Yes, I suppose that is safest.